File: upvpn-android/app/src/main/kotlin/app/upvpn/upvpn/data/db/Device.kt
package app.upvpn.upvpn.data.db

import androidx.room.Entity
import androidx.room.PrimaryKey
import java.util.UUID

@Entity(tableName = "device")
data class Device(
    @PrimaryKey(autoGenerate = false)
    val uniqueId: UUID,
    val name: String,
    val version: String,
    val arch: String,
    val privateKey: String,
    val ipv4Address: String? = null
)

File: upvpn-apple/UpVPN/Model/Device.swift
//
//  Device.swift
//  UpVPN
//
//  Created by Himanshu on 7/16/24.
//

import Foundation

struct Device: Codable {
    var uniqueId: UUID
    var name: String
    var version: String
    var arch: String
    var privateKey: String
    var ipv4Address: String?
    var token: String?
}

extension Device: CustomStringConvertible {
    var description: String {
        let attributes =  [
            "uniqueId: \(self.uniqueId)",
            "name: " + self.name,
            "version: " + self.version,
            "arch: " + self.arch,
            "ipv4Address: " + (self.ipv4Address ?? "nil"),
            "privateKey: <redacted>",
            "token: " + (token.map { _ in "<redacted>" } ?? "nil")
        ].joined(separator: ", ")

        return "Device(\(attributes))"
    }
}
